target/linux/realtek/base-files/etc/board.d/02_network

```diff
@@ -89,6 +91,9 @@ hpe,1920-8g-poe-65w)
 hpe,1920-8g-poe-180w)
 	ucidef_set_poe 180 "$(filter_port_list_reverse "$lan_list" "lan9 lan10")"
 	;;
+hpe,1920-48g-poe)
+	ucidef_set_poe 370 "$(filter_port_list "$lan_list" "lan49 lan50 lan51 lan52")"
```

Comment:
Did you create a proper configuration file, enabling all required ports? By default, the configuration file at `/etc/config/poe` has a limit of 170W set (this is what you get) and only one port enabled.

Reply:
The default file I have is which looks like:
```sh
root@OpenWrt:~# cat /etc/config/poe
config global
        option budget   '385'

config port
        option enable   '1'
        option id       '1'
        option name     'lan1'
        option poe_plus '1'
        option priority '2'
```
I changed it to the following and restarted `/etc/init.d/poe`
```
config global
        option budget   '385'

config port
        option enable   '1'
        option id       '1'
        option name     'lan1'
        option poe_plus '1'
        option priority '2'

config port
        option enable   '1'
        option id       '2'
        option name     'lan2'
        option poe_plus '1'
        option priority '2'

config port
        option enable   '1'
        option id       '3'
        option name     'lan3'
        option poe_plus '1'
        option priority '2'

config port
        option enable   '1'
        option id       '4'
        option name     'lan4'
        option poe_plus '1'
        option priority '2'

config port
        option enable   '1'
        option id       '5'
        option name     'lan5'
        option poe_plus '1'
        option priority '2'

config port
        option enable   '1'
        option id       '6'
        option name     'lan6'
        option poe_plus '1'
        option priority '2'

config port
        option enable   '1'
        option id       '7'
        option name     'lan7'
        option poe_plus '1'
        option priority '2'

config port
        option enable   '1'
        option id       '8'
        option name     'lan8'
        option poe_plus '1'
        option priority '2'

config port
        option enable   '1'
        option id       '9'
        option name     'lan9'
        option poe_plus '1'
        option priority '2'

config port
        option enable   '1'
        option id       '10'
        option name     'lan10'
        option poe_plus '1'
        option priority '2'

config port
        option enable   '1'
        option id       '11'
        option name     'lan11'
        option poe_plus '1'
        option priority '2'

config port
        option enable   '1'
        option id       '12'
        option name     'lan12'
        option poe_plus '1'
        option priority '2'

config port
        option enable   '1'
        option id       '13'
        option name     'lan13'
        option poe_plus '1'
        option priority '2'

config port
        option enable   '1'
        option id       '14'
        option name     'lan14'
        option poe_plus '1'
        option priority '2'

config port
        option enable   '1'
        option id       '15'
        option name     'lan15'
        option poe_plus '1'
        option priority '2'

config port
        option enable   '1'
        option id       '16'
        option name     'lan16'
        option poe_plus '1'
        option priority '2'

config port
        option enable   '1'
        option id       '17'
        option name     'lan17'
        option poe_plus '1'
        option priority '2'

config port
        option enable   '1'
        option id       '18'
        option name     'lan18'
        option poe_plus '1'
        option priority '2'

config port
        option enable   '1'
        option id       '19'
        option name     'lan19'
        option poe_plus '1'
        option priority '2'

config port
        option enable   '1'
        option id       '20'
        option name     'lan20'
        option poe_plus '1'
        option priority '2'

config port
        option enable   '1'
        option id       '21'
        option name     'lan21'
        option poe_plus '1'
        option priority '2'

config port
        option enable   '1'
        option id       '22'
        option name     'lan22'
        option poe_plus '1'
        option priority '2'

config port
        option enable   '1'
        option id       '23'
        option name     'lan23'
        option poe_plus '1'
        option priority '2'

config port
        option enable   '1'
        option id       '24'
        option name     'lan24'
        option poe_plus '1'
        option priority '2'

config port
        option enable   '1'
        option id       '25'
        option name     'lan25'
        option poe_plus '1'
        option priority '2'

config port
        option enable   '1'
        option id       '26'
        option name     'lan26'
        option poe_plus '1'
        option priority '2'

config port
        option enable   '1'
        option id       '27'
        option name     'lan27'
        option poe_plus '1'
        option priority '2'

config port
        option enable   '1'
        option id       '28'
        option name     'lan28'
        option poe_plus '1'
        option priority '2'

config port
        option enable   '1'
        option id       '29'
        option name     'lan29'
        option poe_plus '1'
        option priority '2'

config port
        option enable   '1'
        option id       '30'
        option name     'lan30'
        option poe_plus '1'
        option priority '2'

config port
        option enable   '1'
        option id       '31'
        option name     'lan31'
        option poe_plus '1'
        option priority '2'

config port
        option enable   '1'
        option id       '32'
        option name     'lan32'
        option poe_plus '1'
        option priority '2'

config port
        option enable   '1'
        option id       '33'
        option name     'lan33'
        option poe_plus '1'
        option priority '2'

config port
        option enable   '1'
        option id       '34'
        option name     'lan34'
        option poe_plus '1'
        option priority '2'

config port
        option enable   '1'
        option id       '35'
        option name     'lan35'
        option poe_plus '1'
        option priority '2'

config port
        option enable   '1'
        option id       '36'
        option name     'lan36'
        option poe_plus '1'
        option priority '2'

config port
        option enable   '1'
        option id       '37'
        option name     'lan37'
        option poe_plus '1'
        option priority '2'

config port
        option enable   '1'
        option id       '38'
        option name     'lan38'
        option poe_plus '1'
        option priority '2'

config port
        option enable   '1'
        option id       '39'
        option name     'lan39'
        option poe_plus '1'
        option priority '2'

config port
        option enable   '1'
        option id       '40'
        option name     'lan40'
        option poe_plus '1'
        option priority '2'

config port
        option enable   '1'
        option id       '41'
        option name     'lan41'
        option poe_plus '1'
        option priority '2'

config port
        option enable   '1'
        option id       '42'
        option name     'lan42'
        option poe_plus '1'
        option priority '2'

config port
        option enable   '1'
        option id       '43'
        option name     'lan43'
        option poe_plus '1'
        option priority '2'

config port
        option enable   '1'
        option id       '44'
        option name     'lan44'
        option poe_plus '1'
        option priority '2'

config port
        option enable   '1'
        option id       '45'
        option name     'lan45'
        option poe_plus '1'
        option priority '2'

config port
        option enable   '1'
        option id       '46'
        option name     'lan46'
        option poe_plus '1'
        option priority '2'

config port
        option enable   '1'
        option id       '47'
        option name     'lan47'
        option poe_plus '1'
        option priority '2'

config port
        option enable   '1'
        option id       '48'
        option name     'lan48'
        option poe_plus '1'
        option priority '2'
```
In the status I can now see the first 24 ports:
```sh
root@OpenWrt:~# ubus call poe info
{
        "firmware": "v20.3",
        "mcu": "Nuvoton M05xx LAN Microcontroller",
        "budget": 385.000000,
        "consumption": 0.000000,
        "ports": {
                "lan1": {
                        "priority": 2,
                        "mode": "PoE+",
                        "status": "Searching"
                },
                "lan2": {
                        "priority": 2,
                        "mode": "PoE+",
                        "status": "Searching"
                },
                "lan3": {
                        "priority": 2,
                        "mode": "PoE+",
                        "status": "Searching"
                },
                "lan4": {
                        "priority": 2,
                        "mode": "PoE+",
                        "status": "Searching"
                },
                "lan5": {
                        "priority": 2,
                        "mode": "PoE+",
                        "status": "Searching"
                },
                "lan6": {
                        "priority": 2,
                        "mode": "PoE+",
                        "status": "Searching"
                },
                "lan7": {
                        "priority": 2,
                        "mode": "PoE+",
                        "status": "Searching"
                },
                "lan8": {
                        "priority": 2,
                        "mode": "PoE+",
                        "status": "Searching"
                },
                "lan9": {
                        "priority": 2,
                        "mode": "PoE+",
                        "status": "Searching"
                },
                "lan10": {
                        "priority": 2,
                        "mode": "PoE+",
                        "status": "Searching"
                },
                "lan11": {
                        "priority": 2,
                        "mode": "PoE+",
                        "status": "Searching"
                },
                "lan12": {
                        "priority": 2,
                        "mode": "PoE+",
                        "status": "Searching"
                },
                "lan13": {
                        "priority": 2,
                        "mode": "PoE+",
                        "status": "Searching"
                },
                "lan14": {
                        "priority": 2,
                        "mode": "PoE+",
                        "status": "Searching"
                },
                "lan15": {
                        "priority": 2,
                        "mode": "PoE+",
                        "status": "Searching"
                },
                "lan16": {
                        "priority": 2,
                        "mode": "PoE+",
                        "status": "Searching"
                },
                "lan17": {
                        "priority": 2,
                        "mode": "PoE+",
                        "status": "Searching"
                },
                "lan18": {
                        "priority": 2,
                        "mode": "PoE+",
                        "status": "Searching"
                },
                "lan19": {
                        "priority": 2,
                        "mode": "PoE+",
                        "status": "Searching"
                },
                "lan20": {
                        "priority": 2,
                        "mode": "PoE+",
                        "status": "Searching"
                },
                "lan21": {
                        "priority": 2,
                        "mode": "PoE+",
                        "status": "Searching"
                },
                "lan22": {
                        "priority": 2,
                        "mode": "PoE+",
                        "status": "Searching"
                },
                "lan23": {
                        "priority": 2,
                        "mode": "PoE+",
                        "status": "Searching"
                },
                "lan24": {
                        "priority": 2,
                        "mode": "PoE+",
                        "status": "Searching"
                }
        }
}
```
The access point now powers up from one of the first 24 ports, but the status never changes from `Searching` and also the power consumption does not update. When I booted the stock firmware first the consumption was read correctly.
I can see some errors because of the hardcoded port id limit:
```sh
root@OpenWrt:~# realtek-poe
realtek-poe: invalid port id=25 for lan25realtek-poe: invalid port id=26 for lan26realtek-poe: invalid port id=27 for lan27realtek-poe: invalid port id=28 for lan28realtek-poe: invalid port id=29 for lan29realtek-poe: invalid port id=30 for lan30realtek-poe: invalid port id=31 for lan31realtek-poe: invalid port id=32 for lan32realtek-poe: invalid port id=33 for lan33realtek-poe: invalid port id=34 for lan34realtek-poe: invalid port id=35 for lan35realtek-poe: invalid port id=36 for lan36realtek-poe: invalid port id=37 for lan37realtek-poe: invalid port id=38 for lan38realtek-poe: invalid port id=39 for lan39realtek-poe: invalid port id=40 for lan40realtek-poe: invalid port id=41 for lan41realtek-poe: invalid port id=42 for lan42realtek-poe: invalid port id=43 for lan43realtek-poe: invalid port id=44 for lan44realtek-poe: invalid port id=45 for lan45realtek-poe: invalid port id=46 for lan46realtek-poe: invalid port id=47 for lan47realtek-poe: invalid port id=48 for lan48realtek-poe: Failed to add object: Invalid argument
```
https://github.com/Hurricos/realtek-poe/blob/39c93d39dd10da77b4fe48bc1d6bdd3c5978f866/src/main.c#L116-L119
https://github.com/Hurricos/realtek-poe/blob/39c93d39dd10da77b4fe48bc1d6bdd3c5978f866/src/main.c#L28
I will attempt to recompile with `MAX_PORT 48` but I am not sure if it will fix the status not being read.

Reply:
I tested with `MAX_PORT 48` and can confirm that PoE now works on all ports even while cold booting.
There is still the same issue with reading the status, but that will need to be addressed in realtek-poe.

Reply:
> I tested with `MAX_PORT 48` and can confirm that PoE now works on all ports even while cold booting. There is still the same issue with reading the status, but that will need to be addressed in realtek-poe.

I have opened an issued in realtek-poe to set MAX_PORT to 48. https://github.com/Hurricos/realtek-poe/issues/38

Reply:
@Hurricos I see that a device plugged into lan25 is showing PoE output on lan32 instead of lan25. Any idea why the PoE ports are numbered differently?

Reply:
Some switches have unusual port setup for PoE - e.g. the DGS-1210-28MP series has always 8 ports in reverse. This needs to be taken care of in a mapping.

Reply:
> Some switches have unusual port setup for PoE - e.g. the DGS-1210-28MP series has always 8 ports in reverse. This needs to be taken care of in a mapping.

Where does that mapping live, is there an example elsewhere please?

Reply:
Updated PoE port list to match front panel port names.